wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java

```diff
@@ -136,15 +168,19 @@ protected Connection getValidatedConnection(ConnectionStore store, NamespacedId
    *
    * @param request the http request
    * @param responder the http responder
-   * @param namespace the namespace to check for, or null if no check needs to be performed
+   * @param namespace the namespace to check for
    * @param callable the endpoint logic to run
    */
-  protected <T> void respond(HttpServiceRequest request, HttpServiceResponder responder, @Nullable String namespace,
-                             Callable<T> callable) {
+  protected <T> void respond(HttpServiceRequest request, HttpServiceResponder responder, String namespace,
+                             NamespacedResponder<T> callable) {
     // system namespace does not officially exist, so don't check existence for system namespace.
-    if (namespace != null && !Contexts.SYSTEM.equals(namespace)) {
+    NamespaceSummary namespaceSummary;
+    if (Contexts.SYSTEM.equals(namespace)) {
+      namespaceSummary = new NamespaceSummary(Contexts.SYSTEM, "", 0L);
+    } else {
       try {
-        if (!getContext().getAdmin().namespaceExists(namespace)) {
+        namespaceSummary = getContext().getAdmin().getNamespaceSummary(namespace);
```

Comment:
is this call guaranteed to return latest generation of a namespace? Or only one generation is stored for a namespace at a given time?

Reply:
yep, there is only one at a given time.